### PR TITLE
Another upgrade/update to testing pattern

### LINKED ISSUE #549: Clean up dimension names in PhysicsTestData
The current names are shoc-specific ("shcol" for example). We should use the same approach as p3 to alias these dimension names (via a trivial member function) to a custom name if that's what the client wants.

### PR DESCRIPTION
**Pull request by jgfouca**
1) Allow flexible dimension renaming
2) Have generic, not SHOC-specific, dimension names in the PhysicsTestData base class
3) Add PTD_STD_DEF macro to reduce even more code duplication
4) Fix lots of warnings in shoc tests

Fixes #549

### CODE REVIEW

**Comment by jgfouca** [comment]
For those doing conversions, please rebase/merge your branches once this PR is in (and please, no merges before this PR goes in).

**Comment by jgfouca** [comment]
Recommend viewing with whitespace changes off. We still seem to be introducing a lot of trailing whitespaces in our work; we should try to avoid that. I have my editor set to highlight trailing whitespace in bright red, so I can advise on how to do this in emacs.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 1]
```diff
@@ -210,126 +210,126 @@ void shoc_energy_total_fixer(SHOCEnergytotData &d) {
...
   d.transpose<ekat::util::TransposeDirection::c2f>();
-  compute_conv_vel_shoc_length_c(d.nlev,d.shcol,d.pblh,d.zt_grid,
+  compute_conv_vel_shoc_length_c(d.nlev(),d.shcol(),d.pblh,d.zt_grid,
                                  d.dz_zt,d.thv,d.wthv_sec,d.conv_vel);
   d.transpose<ekat::util::TransposeDirection::f2c>();
 }
 
 void compute_conv_time_shoc_length(SHOCConvtimeData &d) {
-  shoc_init(d.nlev, true);
+  shoc_init(42, true); // fake nlev
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:299]: SHOCConvtimeData doesn't have an nlev, so I assume I just pass whatever int I want?

Comment (bogensch) [reply components/scream/src/physics/shoc/shoc_functions_f90.cpp:299]: Yeah, in these instances (but apparently not this one!) I've been passing it "1" to sort of symbolize a single level.

**Review by bogensch — APPROVED** [review]

File: components/scream/src/physics/share/physics_test_data.hpp [hunk 2]
```diff
@@ -72,25 +74,69 @@ struct SHOCGridData : public PhysicsTestData {
...
+#define PTD_ADD_3_3 6
+#define PTD_ADD_3_4 7
+#define PTD_ADD_3_5 8
+#define PTD_ADD_3_6 9
+#define PTD_ADD_3_7 10
+#define PTD_ADD_3_8 11
+#define PTD_ADD_3_9 12
+
+#define PTD_STD_DEF(name, dim, num_scalars, ...)        \
+  PTD_DATA_COPY_CTOR(name, PTD_ADD_##dim##_##num_scalars);  \
```
Comment (bartgol) [inline components/scream/src/physics/share/physics_test_data.hpp:118]: I know I'm late to the party, but out of curiosity: why did you need the ADD_x_y macro? Didn't something like
```
PTD_DATA_COPY_CTOR(name, (dim + num_scalars) );
```
work?

Comment (bartgol) [reply components/scream/src/physics/share/physics_test_data.hpp:118]: Nvm, I think I understand. The sum of the two needs to be used to append to other macro names, so it must be a simple numeral, not an expression. Got it.
